File: backend/app/modules/dashboard/controller.py
from flask import Blueprint, jsonify, request

from app.modules.dashboard.service import DashboardService
from app.routes.auth import login_required

dashboard_bp = Blueprint("dashboard", __name__, url_prefix="/api/dashboard")
service = DashboardService()


@dashboard_bp.get("/metrics")
@login_required()
def get_metrics(current_user):
    return jsonify(service.get_metrics(request.args.to_dict())), 200

File: backend/src/modules/dashboard/controller.ts
import type { Express } from "express";

import { asyncHandler, getQueryParams } from "../../core/http";
import { loginRequired } from "../../routes/auth";
import { DashboardService } from "./service";

const service = new DashboardService();

export function registerDashboardRoutes(app: Express) {
  app.get(
    "/api/dashboard/metrics",
    loginRequired(),
    asyncHandler(async (req, res) => {
      res.status(200).json(await service.getMetrics(getQueryParams(req.query)));
    }),
  );
}
